src/drivers/imx/sai.c

```diff
@@ -92,8 +97,16 @@ static void sai_start(struct dai *dai, int direction)
 	chan_idx = BIT(0);
 	/* RX3 supports capture on imx8ulp */
 #ifdef CONFIG_IMX8ULP
+	fifo_offset = (dai_fifo(dai, DAI_DIR_PLAYBACK) - dai_base(dai) - REG_SAI_TDR0);
+	if (fifo_offset)
+		fifo_playback = fifo_offset / 4;
+	fifo_offset = (dai_fifo(dai, DAI_DIR_CAPTURE) - dai_base(dai) - REG_SAI_RDR0);
+	if (fifo_offset)
+		fifo_capture = fifo_offset / 4;
```

Comment:
you don't need to calculate `fifo_capture` for playback and `fifo_playback` for capture, right? So maybe move those calculations under respective `if` branches?

Reply:
Hi @lyakh , Thanks for your review. I will update the patches you mentioned.